src/db-api.js

```diff
@@ -3,6 +3,69 @@
 import type { Pool, ResultSet } from 'pg'
 import type { DbApi } from 'icarus-backend'; // eslint-disable-line
 
+/**
+ * Checks if addresses array contains group address. According to Emurgo,
+ * frontend should not explicitly query group addresses, it can lead to hard
+ * to detect bugs.
+ * @param {Db Object} db
+ * @param {Array<Address>} addresses
+ */
+const hasGroupAddress = (db: Pool) => async (
+  addresses: Array<string>,
+): Promise<boolean> => {
+  const res = await db.query({
+    text: 'SELECT EXISTS ( SELECT 1 FROM group_addresses WHERE group_address = ANY($1) )',
+    values: [addresses],
+  })
+  return res.rows[0].exists === true
+}
+
+// NULL value in column POOL means undelegation
+const delegationHistoryQuery = (limit: number) => `
+  SELECT * FROM delegation_certificates
+  WHERE account = $1 
+    AND pool IS NOT NULL
+  ORDER BY block_num DESC LIMIT ${limit}
+`
+
+/**
+ * Returns delegation history for an account
+ * @param {Db Object} db
+ * @param {Account} account
+ */
+const delegationHistoryForAccount = (db: Pool) => async (
+  limit: number,
+  account: string,
+): Promise<ResultSet> =>
+  db.query({
+    text: delegationHistoryQuery(limit),
+    values: [account],
+  })
+
+/**
+* Gets all pools and their details. Currently the tables are joined on pool owner,
+* it bloats the left join a little bit since theres a few pools which even have the same ticker.
```

Comment:
huh? really? pools with same ticker? what is the ticker then good for? :-(

Reply:
It is supposed to be unique according to documentation, although they somehow accepted several pull request with the same ticker